mobly/controllers/android_device.py

```diff
@@ -123,11 +128,16 @@ def _start_services_on_ads(ads):
     for ad in ads:
         running_ads.append(ad)
         try:
-            ad.start_services(skip_sl4a=getattr(ad, "skip_sl4a", False))
-        except:
-            ad.log.exception("Failed to start some services, abort!")
-            destroy(running_ads)
-            raise
+            ad.start_services(skip_sl4a=getattr(ad, KEY_SKIP_SL4A, False))
+        except Exception as e:
+            is_required = getattr(ad, KEY_DEVICE_REQUIRED, True)
+            if is_required:
+                ad.log.exception("Failed to start some services, abort!")
+                destroy(running_ads)
+                raise
+            else:
+                logging.warning("Failed to start some service on %s: %s",
+                                ad.serial, e)
```

Comment:
If you've thought about it and intentionally chose to do it this way, sgtm :)
How about saying "continuing anyway because device is not required" in the message?